superset-frontend/src/explore/components/controls/DatasourceControl.jsx

```diff
@@ -196,6 +207,35 @@ class DatasourceControl extends React.PureComponent {
             </Tooltip>
           </Dropdown>
         </div>
+        {/* missing dataset */}
+        {datasource.id == null && (
+          <div className="error-alert">
+            <ErrorAlert
+              level="warning"
+              title={t('Missing dataset')}
+              source="explore"
+              subtitle={
+                <>
+                  <p>
+                    {t(
+                      'The dataset linked to this chart may have been deleted.',
```

Comment:
Could we be more explicit here as in the other error - "Dataset does not exist".

Reply:
My thinking is this: the dataset does not exist for two reasons, either it never existed or has been deleted. Since the latter is the most likely case, we inform users about this possibility, but also show reservations with "may" to be more accurate.

I think a little bit more color in copy is more helpful comparing to repeating the same general message over and over.